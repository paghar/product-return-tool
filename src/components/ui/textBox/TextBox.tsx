import React from "react";

interface IInputProps {    
    id?: string,
    className?:string,
    value?:string | number ,
    type: "text" | "number" | "textArea" ,
    placeholder?:string,
    onChange?:(e:any) => void,
    name?:string,
}

const TextBox = (props:IInputProps) => {

    const renderElement = props.type === "textArea" 
        ? <textarea 
            id ={props?.id}
            className={props?.className}                 
            value={props?.value}           
            placeholder={props?.placeholder} 
            onChange={props.onChange}
            name={props?.name}  
        >            
        </textarea>
        :  <input 
            id ={props?.id}
            className={props?.className}                 
            value={props?.value}
            type={props?.type}
            placeholder={props?.placeholder} 
            onChange={props.onChange}
            name={props?.name}            
        />;

    return renderElement;
       
    
};

export default TextBox;