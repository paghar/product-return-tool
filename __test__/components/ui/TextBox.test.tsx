import renderer from "react-test-renderer";
import TextBox from "../../../src/components/ui/textBox/TextBox";

describe(">>>> TextBox", () => {
    test("++++ Snapshot TextBox", () => {
        const component = renderer.create(
            <TextBox
                id="txt1"
                value={23}
                type="text"
                placeholder="Insert a number"               
                onChange={jest.fn()}                
            />             
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

export {};