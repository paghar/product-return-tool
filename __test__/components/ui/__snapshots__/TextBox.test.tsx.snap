// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> TextBox ++++ Snapshot TextBox 1`] = `
<input
  id="txt1"
  onChange={[MockFunction]}
  placeholder="Insert a number"
  type="text"
  value={23}
/>
`;
